### DISCUSSION TITLE
Command child process usage

### DISCUSSION THREAD

**Discussion by Blatman** [tauri-apps/tauri#5654] [Q&A] [answered]
Having a devil of a time with these child processes and the scope requirements.

Trying to use the inbuilt plutil on MacOS to convert a plist file to json and it works fine using the Terminal CLI but using the Tauri command api it fails.

The details:
JS:
```js
var cmd = window.__TAURI__.shell.Command

var args = ' -convert json /Users/USER/Documents/HippoLauncher/temp/appPlist.plist -e json';
(note: leading space is required)
var plistData = new cmd('plist2json', [args]).execute()
```
stderr is clear
stdout data shown as -

" -convert json /Users/USER/Documents/HippoLauncher/temp/appPlist.plist -e json: file does not exist or is not readable or is not a regula..."

The appList.plist file definitely does exist in the temp folder and can be converted OK using the same path in the Terminal CLI.

Config:
```json
 "tauri": {
    "allowlist": {
      "all": true,
	  "shell": {
        "execute": true,
        "scope": [
          {
          "name": "plist2json",
          "cmd": "plutil",
	  "sidecar": false,
          "args": true
          }    
	    ]
      },
```

CLI from HippoLauncher dev folder works OK - appPlist.json is generated

`plutil -convert json /Users/USER/Documents/HippoLauncher/temp/appPlist.plist -e json`  // raw cli works OK

Get similar failure trying to use the inbuilt sips utility to extract png data from icns files (works OK with CLI).

Tauri Info: (for hippolauncher dev folder)
```
Environment
  › OS: Mac OS 12.6.0 X64
  › Node.js: 16.14.0
  › npm: 8.19.2
  › pnpm: Not installed!
  › yarn: Not installed!
  › rustup: 1.25.1
  › rustc: 1.63.0
  › cargo: 1.63.0
  › Rust toolchain: stable-aarch64-apple-darwin 

Packages
  › @tauri-apps/cli [NPM]: 1.2.0
  › @tauri-apps/api [NPM]: 1.2.0
  › tauri [RUST]: 1.2.0,
  › tauri-build [RUST]: 1.2.0,
  › tao [RUST]: 0.15.4,
  › wry [RUST]: 0.22.1,

App
  › build-type: bundle
  › CSP: unset
  › distDir: ../dist
  › devPath: ../dist

App directory structure
  ├─ dist
  ├─ node_modules
  └─ src-tauri
```
Fails on both Dev and Prod.

I thought I had the scoping covered and there is no mention of scoping errors that Tauri would normally suggest so not sure why the file is not being seen. Has any one else used either the plutil or sips utilities successfully via the Command API?

**Comment by FabianLars** [comment, marked as answer]
did you try proving `args` as an array instead? Like
```js

var args = ['-convert', 'json', '/Users/USER/Documents/HippoLauncher/temp/appPlist.plist', '-e', 'json'];
var plistData = new cmd('plist2json', args).execute()
```

**Reply by Blatman** [reply]
Ah the magic trick - that works and I see what I was doing wrong, I had tried the array approach earlier but was failing (probably due to another issue) and I ended up mixing strings and arrays.
Thank you - Friday seems a lot better now.
